### PR TITLE
Add unit tests for POSIX clock wrapper

### PR DESCRIPTION
**Pull request by yourslab**
Full coverage for the POSIX clock wrapper

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: platform/posix/utest/clock_utest.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        timeAfterSleep = Clock_GetTimeMs();
+        timeSlept = timeAfterSleep - timeBeforeSleep;
+
+        /* Equality is not asserted because instructions may take longer
+         * to execute depending on the CPU running the test. */
+        TEST_ASSERT_GREATER_OR_EQUAL_UINT32( SLEEP_TIME_MS, timeSlept );
+
+        /* Test that the the time slept is not longer than the elapsed time plus
+         * some tolerance for slower CPUs. */
+        TEST_ASSERT_LESS_OR_EQUAL_UINT32( SLEEP_TIME_MS + SLEEP_TOLERANCE_MS,
```
Comment (leegeth) [inline platform/posix/utest/clock_utest.c:114]: I believe in both the these unit tests, we are dependent on the platform. Can't we mock these platform dependencies to just test the functionality of these 2 functions?

File: platform/posix/utest/clock_utest.c [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+/**
+ * @brief Used to make assertions on the arguments passed to #nanosleep
+ * from #Clock_SleepMs.
+ *
+ * @param[in] __requested_time The requested time to sleep.
+ * @param[in] __remaining The remaining time left to sleep.
+ *
+ * @return Successful status returned by #nanosleep or 0
+ */
+int nanosleep_validate_args( const struct timespec * __requested_time,
```
Comment (leegeth) [inline platform/posix/utest/clock_utest.c:50]: is there a reason why we use the __ added to the param names? If not, can we follow our naming conventions?

File: platform/posix/utest/clock_utest.c [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    timeSpec.tv_nsec = LONG_MAX;
+
+    clock_gettime_ExpectAnyArgsAndReturn( 0 );
+    clock_gettime_ReturnThruPtr___tp( &timeSpec );
+    actualTimeMs = Clock_GetTimeMs();
+
+    expectedTimeMs = ( timeSpec.tv_sec * MILLISECONDS_PER_SECOND )
+                     + ( timeSpec.tv_nsec / NANOSECONDS_PER_MILLISECOND );
+
+    TEST_ASSERT_EQUAL( expectedTimeMs, actualTimeMs );
```
Comment (leegeth) [inline platform/posix/utest/clock_utest.c:114]: I think it is better to add a test case to cover the non overflow case as well.

File: platform/posix/utest/mocks/time_api.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#ifndef TIME_API_
+#define TIME_API_
+
+#include <time.h>
+
+/* Get current value of clock CLOCK_ID and store it in TP.  */
+extern int clock_gettime( clockid_t __clock_id,
```
Comment (leegeth) [inline platform/posix/utest/mocks/time_api.h:7]: same comment about naming.

File: platform/posix/utest/mocks/time_api.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+                          struct timespec * __tp );
+
+/* Pause execution for a number of nanoseconds.
+ *
+ * This function is a cancellation point and therefore not marked with
+ * __THROW.  */
+extern int nanosleep( const struct timespec * __requested_time,
```
Comment (leegeth) [inline platform/posix/utest/mocks/time_api.h:14]: naming here as well

**Review by leegeth — DISMISSED** [review]

**Review by leegeth — DISMISSED** [review]

File: platform/posix/utest/mocks/time_api.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#ifndef TIME_API_
+#define TIME_API_
+
+#include <time.h>
+
+/* Get current value of clock CLOCK_ID and store it in TP.  */
```
Comment (gkwicker) [inline platform/posix/utest/mocks/time_api.h:6]: nit: rename `tp` to something longer (e.g. `timepoint`) so that we don't have to add a 2-character string which could easily be a misspelled word to the lexicon.  The parameter's name doesn't have to match the original definition, only the type.

**Review by gkwicker — CHANGES_REQUESTED** [review]
See comment

File: platform/posix/utest/reconnect_utest.c [hunk 7]
```diff
@@ -67,6 +67,7 @@ static void verifyReconnectParamsAfterReset( void )
  */
 void test_Transport_ReconnectParamsReset_Sets_Jitter_Correctly( void )
 {
+    clock_gettime_ExpectAnyArgsAndReturn( 0 );
```
Comment (yourslab) [inline platform/posix/utest/reconnect_utest.c:70]: I need to expect these now in the reconnect unit tests as they are being mocked.

**Review by leegeth — APPROVED** [review]

**Review by gkwicker — APPROVED** [review]
